homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,214 @@
+        ]
+
+    async_add_entities(await hass.async_add_executor_job(get_thermostats))
+
+
+class SomfyClimate(SomfyEntity, ClimateEntity):
+    """Representation of a Somfy thermostat device."""
+
+    def __init__(self, coordinator, device_id, api):
+        """Initialize the Somfy device."""
+        super().__init__(coordinator, device_id, api)
+        self._climate = None
+        self._create_device()
+
+    def _create_device(self):
+        """Update the device with the latest data."""
+        self._climate = Thermostat(self.device, self.api)
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the state attributes of the device."""
+        return {ATTR_BATTERY_LEVEL: self._climate.get_battery()}
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._climate.get_ambient_temperature()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._climate.get_target_temperature()
+
+    def set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self._climate.set_target(TargetMode.MANUAL, temperature, DurationType.NEXT_MODE)
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 26.0
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 15.0
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._climate.get_humidity()
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if self._climate.get_regulation_state() == RegulationState.TIMETABLE:
+            return HVAC_MODE_AUTO
+        return HVAC_MODES_MAPPING.get(self._climate.get_hvac_state())
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes.
+
+        HEAT and COOL mode are exclusive. End user has to enable a mode manually within the Somfy application.
+        So only one mode can be displayed. Auto mode is a scheduler.
+        """
+        hvac_state = HVAC_MODES_MAPPING.get(self._climate.get_hvac_state())
+        return [HVAC_MODE_AUTO, hvac_state]
+
+    def set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == self.hvac_mode:
```

Comment:
We try avoid these kinds of guards since they make it impossible to change state if we have incorrect state.

Reply:
OK. So you mean that's because of this safe guard we Won't call the api and fix and invalid state?

Reply:
We return if the case is true so we can't change device state if we have incorrect state.